setuptools/tests/test_build_meta.py

```diff
@@ -981,3 +982,99 @@ def test_system_exit_in_setuppy(monkeypatch, tmp_path):
     with pytest.raises(SystemExit, match="some error"):
         backend = BuildBackend(backend_name="setuptools.build_meta")
         backend.get_requires_for_build_wheel()
+
+
+class TestTypeInformationIncludedByDefault:
+    dont_include_package_data = """
+    [project]
+    name = "foo"
+    version = "1"
+    [tools.setuptools]
+    include-package-data = false
+    """
+
+    exclude_type_info = """
+    [tool.setuptools.exclude-package-data]
+    "*" = ["py.typed", "*.pyi"]
+    """
+
+    package_1 = {
+        "foo": {
+            "bar.pyi": "",
+            "py.typed": "",
+        }
+    }
+
+    package_2 = {
+        "foo": {
+            "bar": {
+                "py.typed": "",
+                "mod.pyi": "",
+            }
+        }
+    }
+
+    package_3 = {
+        "foo": {
+            "namespace": {
+                "foo.pyi": "",
+            },
+            "__init__.pyi": "",
+            "py.typed": "",
+        }
+    }
+
+    packages_to_test = [package_1, package_2, package_3]
+
+    def is_type_information_file(self, filename):
+        basename = os.path.basename(filename)
+        return basename.endswith(".pyi") or basename == "py.typed"
+
+    def get_type_files(self, file_spec):
+        output = set()
+        for key in file_spec.keys():
+            if isinstance(file_spec[key], str):
+                if self.is_type_information_file(key):
+                    output.add(key)
+            else:
+                output.update(
+                    key + "/" + f for f in self.get_type_files(file_spec[key])
+                )
+        return output
+
+    @pytest.fixture(params=itertools.product(packages_to_test, [True, False]))
```

Comment:
`itertools.product` is an interesting detail, it seems that that we have a large amount of tests being produced...

Using `build_backend.build_wheel(dist_dir)` in the `test_build_meta.py` file is very expensive and increases the total time necessary for running the tests (the faster the test suite runs the better it is for the developer's perspective).

I wonder if it would be better to use less expensive tests in `test_build_py`[^2]? If that is not enough to get confidence, we can keep 1 heavy test in `test_build_meta.py` and adopt light tests for the edge cases in `test_build_py`...

What do you think?

[^2]: Maybe test `find_data_files` directly (that is the "leanest" test I can think of) or alternatively check `build_py.get_outputs()`

Reply:
I removed the existing test from test_build_meta and instead simply edited one of the existing tests. Is this ok to do?